Description: Set the SONAME properly
Author: Fabian Wolff <fabi.wolff@arcor.de>
Last-Update: 2020-01-08
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -125,8 +125,8 @@
   # libz3.so.W.X.
   # This indicates that no breaking API changes will be made within a single
   # minor version.
-  VERSION ${Z3_VERSION}
-  SOVERSION ${Z3_VERSION_MAJOR}.${Z3_VERSION_MINOR})
+  VERSION 4
+  SOVERSION 4)
 
 if (NOT MSVC)
   # On UNIX like platforms if we don't change the OUTPUT_NAME
--- a/src/api/java/CMakeLists.txt
+++ b/src/api/java/CMakeLists.txt
@@ -52,6 +52,7 @@
 
 # This prevents CMake from automatically defining ``z3java_EXPORTS``
 set_property(TARGET z3java PROPERTY DEFINE_SYMBOL "")
+set_property(TARGET z3java PROPERTY NO_SONAME ON)
 
 # Rule to generate the ``com.microsoft.z3.enumerations`` package
 # FIXME: This list of files is fragile
@@ -223,7 +224,7 @@
   )
   # FIXME: I don't think this the right installation location
   set(Z3_JAVA_JNI_LIB_INSTALLDIR
-    "${CMAKE_INSTALL_LIBDIR}"
+    "${CMAKE_INSTALL_LIBDIR}/jni"
     CACHE
     PATH
     "Directory to install Z3 Java JNI bridge library relative to install prefix"
